src/linear_algebra/basic.lean

```diff
@@ -806,6 +807,20 @@ begin
   simp only [Sup_eq_supr', mem_supr_of_directed _ hdir.directed_coe, set_coe.exists, subtype.coe_mk]
 end
 
+@[norm_cast, simp] lemma coe_supr_of_chain (a : ℕ →ₘ submodule R M) :
+  (↑(⨆ k, a k) : set M) = ⋃ k, (a k : set M) :=
+coe_supr_of_directed a a.monotone.directed_le
+
+/-- We can regard `coe_supr_of_chain` as the statement that `coe : (submodule R M) → set M` is
+Scott continuous for the ω-complete partial order induced by the complete lattice structures. -/
+def coe_scott_continuous : submodule R M →𝒄 set M :=
+{ to_fun    := coe,
+  monotone' := set_like.coe_mono,
+  cont      := coe_supr_of_chain, }
```

Comment:
@cipher1024: which of the many bundled and unbundled forms is better here? Another option would be:
```suggestion
/-- We can regard `coe_supr_of_chain` as the statement that `coe : (submodule R M) → set M` is
Scott continuous for the ω-complete partial order induced by the complete lattice structures. -/
lemma coe_scott_continuous : omega_complete_partial_order.continuous' (coe : submodule R M → set M) :=
⟨set_like.coe_mono, coe_supr_of_chain⟩
```

Reply:
I haven't been too involved with the bundling / unbundling debate so maybe @semorrison could enlighten us on the matter. For these files, the code reviewers were suggesting I go for a more bundled approach so that's the direction I favored. That being said, I think it depends on how you're going to use it. If the fact that it's a continuous homomorphism is only one of multiple facets of this coercion, unbundled will allow you to make the various aspect intervene with equal ease. If you only use it a continuous homomorphism, bundled makes it easy to trigger the right theorems.

Reply:
For what it's worth, I'm not entirely convinced we need to include an encoding of `coe_supr_of_chain` in the language of `omega_complete_partial_order`.

For one thing, having `order.omega_complete_partial_order` as an import to `linear_algebra.basic` will make it rather painful for anyone developing `omega_complete_partial_order` in the future. Since I also don't think `order.omega_complete_partial_order` should import `linear_algebra.basic`, and I think this tiny result does not justify the creation of a new file, I think we could just drop `coe_scott_continuous` and move on.

Reply:
I hasten to say, thanks for the input @cipher1024 and also thanks for authoring `omega_complete_partial_order` from which I learned about ω-complete partial orders!

Reply:
I like Eric's suggestion, and as this PR is long delayed I just went ahead and merged it. I think we can deal with unentangling the imports separately.